### ISSUE TITLE
IM 7.0.10-46 increases time to identify significantly

### ISSUE THREAD

**Issue by dexbleeker** [ImageMagick/ImageMagick#3183]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of ImageMagick (It is still a problem in 7.0.10-61)
- [x] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
When identifying images/pdf-files, before version 7.0.10-46 ImageMagick would always finish in less than a second. Therefore we always use `-limit time 1`. Since the upgrade to 7.0.10-46 (or later), this leads to errors (`identify: time limit exceeded 'Operation canceled' @ fatal/cache.c/GetImagePixelCache/1716.`) for some files.
Because the time needed increases with a factor ten for some files, I assume this is a bug.

### Steps to Reproduce
To not mess with my ImageMagick install locally, I've tested this in Docker (with versions 45 and 51 as 46-50 are not available from this particular image):
`docker run --rm --entrypoint=identify -v <abspath>:/imgs dpokidov/imagemagick:7.0.10-<45 or 51>-buster -limit time 1 -limit thread 1 /imgs/<filename>`
With [100_pages.pdf](https://github.com/ImageMagick/ImageMagick/files/5911798/100_pages.pdf), version 45 completes succesfully:
```
/imgs/100_pages.pdf[0] PDF 595x842 595x842+0+0 16-bit sRGB 12028B 0.030u 0:00.030
/imgs/100_pages.pdf[1] PDF 595x842 595x842+0+0 16-bit sRGB 7217B 0.030u 0:00.033
/imgs/100_pages.pdf[2] PDF 595x842 595x842+0+0 16-bit sRGB 7549B 0.030u 0:00.033
<snip>
/imgs/100_pages.pdf[99] PDF 595x842 595x842+0+0 16-bit sRGB 8112B 0.010u 0:00.027
```
Version 51 however, takes much longer per page and eventually reaches the limit and exits.

The differences in speed become very clear and are extreme when a 2MB heic file is used (GH won't allow .heic files so here is a zip with the file: [heic-file.zip](https://github.com/ImageMagick/ImageMagick/files/5911850/heic-file.zip)):

* Version 45: `/imgs/example.heic HEIC 4032x3024 4032x3024+0+0 8-bit YCbCr 0.010u 0:00.035`
* Version 51: `/imgs/example.heic HEIC 4032x3024 4032x3024+0+0 8-bit YCbCr 1.420u 0:00.930`

A very significant increase indeed.

Is there some option for `identify` to improve the performance again? Is this in fact a bug, or is this intended and will all versions from 46 have a significant performance drop for identifying files?

### System Configuration
Tested on both Linux and with the ImageMagick installed in MacOS with homebrew. For Linux all versions from 7.0.10-45 are tested  (in addition to 45 and 51 from the Docker steps above) and MacOS was tested version 7.0.10-61.

**Comment by urban-warrior** [comment]
We're using ImageMagick 7.0.10-62 Beta and the `identify 100_pages.pdf` command completed instantly (0:00.002).  Your HEIC file completed instantly as well (0:00.003).  Not sure why its lagging for you.

**Comment by dexbleeker** [comment]
I've tested with 7.0.10-62 Beta and unfortunately I'm still having problems. It is weird to me that it consistenly lags with version 46 and above, but it is nice to know that it is working for you. Is the configuration output helpful maybe? 
What else can I do to debug this problem? 
Thank you for your time.
```
==============================================================================
ImageMagick 7.0.10-62 is configured as follows. Please verify that this
configuration matches your expectations.

Host system type: x86_64-pc-linux-gnu
Build system type: x86_64-pc-linux-gnu

                  Option                        Value
------------------------------------------------------------------------------
Shared libraries  --enable-shared=yes           yes
Static libraries  --enable-static=yes           yes
Build utilities   --with-utilities=yes          yes
Module support    --with-modules=no             no
GNU ld            --with-gnu-ld=yes             yes
Quantum depth     --with-quantum-depth=16       16
High Dynamic Range Imagery
                  --enable-hdri=yes             yes

Install documentation:                          yes

Memory allocation library:
  JEMalloc          --with-jemalloc=no          no
  TCMalloc          --with-tcmalloc=no          no
  UMem              --with-umem=no              no

Delegate library configuration:
  BZLIB             --with-bzlib=yes            no
  Autotrace         --with-autotrace=no         no
  DJVU              --with-djvu=yes             no
  DPS               --with-dps=yes              no
  FFTW              --with-fftw=yes             no
  FLIF              --with-flif=yes             no
  FlashPIX          --with-fpx=yes              no
  FontConfig        --with-fontconfig=yes       yes
  FreeType          --with-freetype=yes         yes
  Ghostscript lib   --with-gslib=no             no
  Graphviz          --with-gvc=yes              no
  HEIC              --with-heic=yes             yes
  JBIG              --with-jbig=yes             yes
  JPEG v1           --with-jpeg=yes             yes
  JPEG XL           --with-jxl=no               no
  LCMS              --with-lcms=yes             no
  LQR               --with-lqr=yes              no
  LTDL              --with-ltdl=no              no
  LZMA              --with-lzma=yes             yes
  Magick++          --with-magick-plus-plus=yes yes
  OpenEXR           --with-openexr=yes          no
  OpenJP2           --with-openjp2=yes          no
  PANGO             --with-pango=yes            no
  PERL              --with-perl=no              no
  PNG               --with-png=yes              yes
  RAQM              --with-raqm=yes             no
  RAW               --with-raw=yes              no
  RSVG              --with-rsvg=yes             yes
  TIFF              --with-tiff=yes             yes
  WEBP              --with-webp=yes             yes
  WMF               --with-wmf=no               no
  X11               --with-x=                   yes
  XML               --with-xml=yes              yes
  ZIP               --with-ziplib=yes           no
  ZLIB              --with-zlib=yes             yes
  ZSTD              --with-zstd=yes             no

Delegate program configuration:
  GhostPCL          None                        pcl6 (unknown)
  GhostXPS          None                        gxps (unknown)
  Ghostscript       None                        gs (9.53.3)

Font configuration:
  Apple fonts       --with-apple-font-dir=default
  Dejavu fonts      --with-dejavu-font-dir=default      none
  Ghostscript fonts --with-gs-font-dir=default          /usr/local/share/ghostscript/fonts/
  URW-base35 fonts  --with-urw-base35-font-dir=default  none
  Windows fonts     --with-windows-font-dir=default     none

X11 configuration:
  X_CFLAGS        =
  X_PRE_LIBS      =  -lSM -lICE
  X_LIBS          =
  X_EXTRA_LIBS    =

Options used to compile and link:
  PREFIX          = /usr/local
  EXEC-PREFIX     = /usr/local
  VERSION         = 7.0.10-62
  CC              = gcc
  CFLAGS          = -I/usr/include/libxml2   -I/usr/local/include/libpng16 -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -pthread -I/usr/local/include/libpng16 -I/usr/include/librsvg-2.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/cairo -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/local/include/libpng16  -I/usr/local/include -I/usr/local/include/libpng16 -I/usr/include/freetype2 -I/usr/local/include/libpng16 -I/usr/include/uuid -I/usr/include/freetype2  -fopenmp -Wall -g -O2 -mtune=core2 -fexceptions -pthread -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16
  CPPFLAGS        =  -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16
  PCFLAGS         =
  DEFS            = -DHAVE_CONFIG_H
  LDFLAGS         =
  LIBS            =  -ljbig  -ltiff -lfreetype  -ljpeg    -L/usr/local/lib -lpng16 -lz      -lfontconfig -lfreetype -L/usr/local/lib -lheif -lwebpmux -lwebpdemux -lwebp -lwebp   -lXext    -lSM -lICE -lX11  -llzma      -lrsvg-2 -lm -lgio-2.0 -lgdk_pixbuf-2.0 -lgobject-2.0 -lglib-2.0 -lcairo -lxml2  -lz -lz    -lm    -lpthread
  CXX             = g++
  CXXFLAGS        =  -pthread
  FEATURES        = DPC HDRI Cipher OpenMP
  DELEGATES       = fontconfig freetype heic jbig jng jpeg lzma png ps rsvg tiff video webp x xml zlib
==============================================================================

```

**Comment by RemcodM** [comment]
I did some research to this issue as I was able to reproduce the issue.

It seems the problem is caused by commit e6ab1e8b4c377f51898e0af6c0dceced89dc39cd. This changes the behaviour of the `identify` command such that it always 'pings' (reads) the entire image file when the `-format` option is used. We were using that option, which was causing the problem.

Luckily, we can just set `-ping` as option to the `identify` command to revert this behaviour and only read the simple properties we need.

The reason @urban-warrior did not see any increase between those versions is probably because no `-format` parameter was used. Therefore, a different default was used in both versions, which resulted in the same behaviour under both versions.
